apps/examples/testcase/kernel: Add new TC on sysdbg

deepaksrma opened the pull request:
- add new TCs to test sysdbg APIs

Signed-off-by: Deepak Sharma <deepak.sh@samsung.com>

sangam-swami commented on apps/examples/testcase/le_tc/kernel/tc_debug.c:
```diff
@@ -0,0 +1,104 @@
+
+	strcpy(buffer, "read");
+	ret = write(fd, buffer, sizeof(buffer));
+	TC_ASSERT_EQ_CLEANUP("write", ret, 0, close(fd));
```
here ret contains 0 for success case?

sangam-swami commented on apps/examples/testcase/le_tc/kernel/Kconfig:
```diff
@@ -22,6 +22,7 @@ config TC_KN_ALL
 	select TC_KERNEL_CLOCK
 	select TC_KERNEL_ENVIRON
 	select TC_KERNEL_ERRNO
+	select TC_KERNEL_DEBUG if DEBUG_SYSTEM
```
can you update the tc log in the commit message

Commit:
build/configs: add the changes in defconfig for sysdbg TC

- add CONFIG_TC_KERNEL_DEBUG in defconfigs to test sysdbg TC

Signed-off-by: Deepak Sharma <deepak.sh@samsung.com>

Commit:
os/kernel/sysdbg: add the macro for sysdbg path

- add the macro for sysdbg path

Signed-off-by: Deepak Sharma <deepak.sh@samsung.com>

Commit:
apps/examples/testcase/kernel: Add new TC on sysdbg

- add new TCs to test sysdbg APIs

logs for this TC are:
sysdbg_monitor_enab: Enabled sysdbg monitoring feature
sysdbg_print: Dilaying the TASK SCHEDULING HISTORY for 32 count
sysdbg_print:****************************************************************************
sysdbg_print: * TASK_SCHEDTIME          TASK_NAM PID     PRIORITY        TCB
sysdbg_print: ***************************************************************************
sysdbg_print:     4136                      kernel_tc         9      80         2004040
sysdbg_print:     4136                    kernel_tc         9       180         2004040
sysdbg_print:     4134                           tash         6       1         2004840
sysdbg_print:     4101                      le Task         0         0        1003DB98
sysdbg_print:     4101                      Idle Task         0         0      1003DB98
sysdbg_print:     4101                    LWIPCP/IP         3       105        100767A0
sysdbg_print:     4001                      Idle Task         0         0      1003DB98
sysdbg_print:     4001                      Idle sk         0         0        1003DB98
sysdbg_print:     4001                    LWIP_TCP/IP         3       105      100767A0
sysdbg_print:     3967                      Idle Ta         0         0        1003DB98
sysdbg_print:     3967                      Idle Task         0         0       003DB98
sysdbg_print:     3966                           tash       6       125         2004840
sysdbg_print:     3901                      Idle Task         0         0        10DB98
sysdbg_print:     3901                      Idle Task       0         0        1003DB98
sysdbg_print:     3901                    LWIP_TCP/IP         3       105        1007A0
sysdbg_print:     3862                      Idle Task       0         0        1003DB98
sysdbg_print:     3862                      Idle Task         0         0        1003DB
sysdbg_print:     3861                           tash       6       125         2004840
ysdbg_print:     3801                      Idle Task         0       0        1003DB98B98
sysdbg_print:     3801                    LWIP_TCP/IP         3       105        100767A0
sdbg_print:     3701                      Idle Task         0       0        1003DB98
sysdbg_print:     3701                      Idle Task         0         0        1003DB98
sysg_print:     3701                    LWIP_TCP/IP         3     105        100767A0
sysdbg_print:     3601                      Idle Task         0         0        1003DB98
sysdbgrint:     3601                      Idle Task         0       0        1003DB98
sysdbg_print:     3601                    LWIP_TCP/IP         3       105        100767A0
sysdbg_pnt:     3583                      Idle Task         0                1003DB98
sysdbg_print:     3583                      Idle Task         0         0        1003DB98
sysdbg_pri:     3582                           tash         6       125       2004840
sysdbg_print:     3501                      Idle Task         0         0        1003DB98
sysdbg_print:   3501                      Idle Task         0         0      1003DB98
sysdbg_print: Displaying the IRQ SCHEDULING HISTORY for 32 count
sysdbg_print: ***************************************************************************
sysdbg_prin * IRQ_TIME   IRQ_NUMBER      ISR_ADDRESS
sysdbg_print: ***************************************************************************
sysdbg_print:     4143        15         E00D9CD
sysdbg_prt:     4142          15         E00D9CD
sysdbg_print:     4141        15         E00D9CD
sysdbg_print:     4140        15         E00D9CD
sysg_print:     4139          15         E00D9CD
sysdbg_print:     4138      15   E00D9CD
sysdbg_print:     4138        19        10013A35
sysdbg_print:     4137        15         E00D9CD
sysdbg_print:     41          19        10013A35
sysdbg_print:     4137        19        10013A35
sdbg_print:     4136          15         E00D9CD
sysdbg_print:     4136        19        10013A35
sysdbg_print:     4136        11        1001BD
sysdbg_print:     4135        15         E00D9CD
sysdbg_print:   4135          19        10013A35
sysdbg_print:     4135        19        10013A35
sysdbg_print:     4134        15         E00D9CD
sysdbg_prin     4134          19        10013A35
sysdbg_print:     4133        15         0D9CD
sysdbg_print:     4132        15         E00D9CD
sysdbg_print:     4131        15         E00D9CD
sysdbg_print:     4130      15   E00D9CD
sysdbg_print:     4129        15         E00D9CD
sysdbg_int:     4128          15         E00D9CD
sysdbg_print:     4127        15         E00D9CD
sysdbg_print:     4126        15         E00D9CD
sdbg_print:     4125          15         E00D9CD
sysdbg_print:     4124      15   E00D9CD
sysdbg_print:     4123        15         E00D9CD
sysdbg_print:     4122        15         E00D9CD
sysdbg_print:     21          15         E00D9CD
sysdbg_print:     4120        15         E00D9CDsysdbg_print: Displaying the SEMAPHORE HISTORY for 32 count
sysdbg_print: IN:INIT AQ:AQUIRED RL:RELEASED WT:WAITING D DESTROY UN:UNKNOWN
sysdbg_print: PID : -1 indicates that, it a ISR, for ISR, TCB field gives reference of ISR
sysdbg_print: *******************************************************************************
sysdbg_print: SEMAPHORE_TIME     STAT             TASK_NAME      SEMAPHORE     TCB       PID
sysdbg_print: *******************************************************************************
sysdbg_print:     4138                RL         kernel_tc         20041A0     2004040     9
sysdbg_print:     4138                AQ           kernel_tc         201A0     2004040     9
sysdbg_print:     4138                R          kernel_tc        100401DC     2004040     9
sysdbg_print:     4138                AQ           kernel_tc      100401DC     2004040     9
sysdbg_print:     4138              RL           kernel_tc         20041A0     2004040     9
sysdbg_print:     4138                AQ           kernetc         20041A0     2004040     9
sysdbg_print:     4138              RL           kernel_tc        100401DC     2004040     9
sysdbg_print:     4138                AQ         kernel_tc        100401DC     2004040     9
sysdbg_print:   4138                RL           kernel_tc         2004254     2004040     9
sysdbg_print:     4138                RL         kernel_tc        1003D400     2004040     9
sysdbg_pnt:     4137                AQ           kernel_tc        1003D400     2004040     9
sysdbg_print:     4137              RL           kernel_tc        1003D400     2004040     9
sdbg_print:     4136                AQ           kernel_tc        1003D400     2004040     9
sysdbg_print:     4136              AQ           kernel_tc         2004254     2004040   9
sysdbg_print:     4136                AQ           kernel_tc        10041C0C     2004040     9
sysdbg_print:    136                IN           kernel_tc        10041C0C     04040     9
sysdbg_print:     4136                RL                tash        10041884     2004840     6
sysdbg_pri:     4136                AQ                tash        100418     2004840     6
sysdbg_print:     4136                RL                tash         2004224     2004840     6
sybg_print:     4136                RL                tash      10041884     2004840     6
sysdbg_print:     4136                AQ                tash        10041884     2004840   6
sysdbg_print:     4136                IN                ta         200427C     2004840     6
sysdbg_print:     4136                AQ                tash         2004224     24840     6
sysdbg_print:     4136                RL              tash         2004224     2004840     6
sysdbg_print:     4136                RL                tash        1004188    2004840     6
sysdbg_print:     4136                AQ              tash        10041884     2004840     6
sysdbg_print:     4136                IN                tash       2004254     2004840     6
sysdbg_print:     4136              AQ                tash         2004224     2004840     6
sysdbg_print:     4136                RL                tas        2004224     2004840     6
sysdbg_print:     4136              RL                tash        10041884     2004840     6
sysdbg_print:     4136                AQ              tash        10041884     2004840     6
sysdbg_print:     41                IN                tash         200422C     2004840     6
sysdbg_write: To use this feature, kindly enle CONFIG_FRAME_POINTER
sysdbg_write: To use this feature, kily enable CONFIG_FRAME_POINTER
sysdbg_monitor_disable: Disabled sysdbg monitoring feature

[_debug_sysdbg_write] PASS

Signed-off-by: Deepak Sharma <deepak.sh@samsung.com>
